src/utils.ts

```diff
@@ -0,0 +1,31 @@
+import {decode, encode} from "iconv-lite";
+
+export function wrap(text: string, wrapAt: number): string {
+    let result = "";
+    let sep = ""
+    for(let i = 0; i < text.length; i += wrapAt) {
+        result += sep + text.substring(i, Math.min(i + wrapAt, text.length));
+        sep = "\n";
+    }
+    return result;
+}
+
+export function base64Encode(text: string, encodingName: string): string {
+    try {
+        const data: Buffer = encode(text, encodingName);
+        const decodedString = decode(data, encodingName);
+        if (text != decodedString) {
+            return `Input contains characters outside of ${encodingName}`;
+        }
+        const binaryString = data.toString("binary");
+        return btoa(binaryString);
+    } catch (e) {
+        return `Input contains characters outside of ${encodingName}`;
+    }
+}
+
+export function base64EncodeLinesSeparately(textInputValue: string, encodingName: string): string {
```

Comment:
I'd prefer for the first method in the chain to also be on the next line like
```
    const result = textInputValue
        .split(/\r\n|\r|\n/)
        .map((line) => base64Encode(line, encodingName))
        .join('\n');
    return result;
```

Problem with that is that the method calls are further left that the object `textInput`value.
However you can't just add a newline after the return, because that would return undefined.
Is this overkill?

```
    return (
        textInputValue
            .split(/\r\n|\r|\n/)
            .map(line => base64Encode(line, encodingName))
            .join("\n")
    );
```

Reply:
I totally get your point and I agree that the current format is pretty hard to read. How about we activate 'align when multiline' for chained method calls in the code style? Or would you prefer the first method call to be separated from 'textInputValue'?

Reply:
* first of all it would be nice if the first method call is aligned to all the other method calls
* previously I favored the alignment, but it has two disadvantages:
1) it depends on the length of the first line (if I rename textInputValue) then following lines would also need re-indenting, even though the have not changed)
2) indenting gets very large pretty soon, which may require additional line breaks

I think I'd like to try without alignment and instead a newline before the first method call (which unfortunately requires the additional parentheses)

Reply:
Done